workspace/__lib__/xod/datetime/pad-with-zeroes/patch.cpp

```diff
@@ -0,0 +1,30 @@
+struct State {
+    char str[16];
+    CStringView view;
+    State()
+        : view(str) {}
+};
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    auto state = getState(ctx);
+
+    auto num = (getValue<input_IN>(ctx) == 0) ? 0 : abs(getValue<input_IN>(ctx));
+    int8_t lenFull = (getValue<input_W>(ctx) < 0) ? 0 : min(15, getValue<input_W>(ctx));
+
+    char strNum[16];
+    dtostrf(num, 0, 0, strNum);
+
+    CStringView numView(strNum);
+    int8_t lenStr = length(XString(&numView));
```

Comment:
:open_mouth: this can be really made more simple:

```cpp
int8_t lenStr = strlen(strNum);
```

No `CStringView`/`XString` magic is required